### PR TITLE
Reverse API changes to support :first as default and add opts to parse_http

### LINKED ISSUE #60: Remote parser does not have natures/formats/returns options

### PR DESCRIPTION
**Pull request by bermannoah**
Closes #60

### CODE REVIEW

**Comment by bermannoah** [comment]
@julik switched to using `:first` as default rather than always returning an array. Feel like it makes sense to have "return all the things" be optional rather than enforced?

File: spec/remote_fetching_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 require 'webrick'
 
-describe 'Fetching data from HTTP remotes' do
+describe 'Fetching data from HTTP remotes' do
```
Comment (julik) [inline spec/remote_fetching_spec.rb]: Stray space

File: spec/format_parser_spec.rb [hunk 2]
```diff
@@ -26,24 +26,24 @@
...
 
-      subject { FormatParser.parse(blob) }
+      subject { FormatParser.parse(blob, results: :all) }
 
       it { is_expected.to include(image) }
       it { is_expected.to include(audio) }
     end
-
-    context '#parse called with hash options' do
+\
```
Comment (julik) [inline spec/format_parser_spec.rb]: Stray slash

**Comment by julik** [comment]
@bermannoah There is also a fuzzing error in the MP3 parser - it can be reproduced on this branch, see https://travis-ci.org/WeTransfer/format_parser/jobs/332196495

**Comment by bermannoah** [comment]
@julik this is a weird one. Think it's coming from the read method in IOWrapper. Pushed a fix, let me know. :)

**Pull request merged**